chain/sync.go

```diff
@@ -529,7 +529,24 @@ func blockSanityChecks(h *types.BlockHeader) error {
 }
 
 // Should match up with 'Semantical Validation' in validation.md in the spec
-func (syncer *Syncer) ValidateBlock(ctx context.Context, b *types.FullBlock) error {
+func (syncer *Syncer) ValidateBlock(ctx context.Context, b *types.FullBlock) (err error) {
+	defer func() {
+		// b.Cid() could panic for empty blocks that are used in tests.
```

Comment:
@whyrusleeping Can you quickly confirm you're okay with this recover being added?

Reply:
Its probably fine. The better approach is to guard against it by checking for the empty condition that would cause b.Cid() to panic, but in lieu of that, the defer works well enough.